### ISSUE TITLE
[Bug] Custom Renderer: overridden method not called

### ISSUE THREAD

**Issue by Poli97** [xamarin/Xamarin.Forms#9397]
### Description
Context (in case you are not familiar with screen-readers). Screen-readers are available on iOS (VoiceOver) and Android (TalkBack) to support accessibility to people with sever visual impairments. In a nutshell, when the screen-reader is activated and the user touches a UI elements, it gets the “accessibility focus" (the “on tap” event is not fired) and then, when the user double taps, the “on tap” event is fired. In native iOS and Android code it is possible to override methods to specify what to do when a UI element gets or loses the accessibility focus. The methods are called [OnPopulateAccessibilityEvent()](https://docs.microsoft.com/en-us/dotnet/api/android.views.view.onpopulateaccessibilityevent?view=xamarin-android-sdk-9) for Android and [AccessibilityElementDidBecomeFocused()](https://docs.microsoft.com/en-us/dotnet/api/uikit.uiresponder.accessibilityelementdidbecomefocused?view=xamarin-ios-sdk-12) for iOS. (Also official [Android Java](https://developer.android.com/reference/android/support/v4/view/AccessibilityDelegateCompat#onpopulateaccessibilityevent) and [iOS Swift](https://developer.apple.com/documentation/objectivec/nsobject/1615183-accessibilityelementdidbecomefoc) documentation).

Problem 1. I couldn’t find a method in Xamarin Forms to manage the accessibility focus events (when the UI element gets or loses the accessibility focus). Do you know if such methods exist in Xamarin Forms? In case they don’t, we believe that they are very important for the development of accessibile applications, so we would like to suggest to add them.

Problem 2. In order to overcome Problem 1, I am trying to make a Custom Renderer that uses native buttons. The custom rendered seems to be correctly initialized (the right message is logged) and shown on the screen. However, the overridden methods are not called. Code is available [My Repo](https://github.com/Poli97/CustomViewAccessibility). Am I doing something wrong?

### Steps to Reproduce

1. Create a Custom Control
2. Create the Custom Renderer on each platform
3. Override OnPopulateAccessibilityEvent for Android and AccessibilityElementDidBecomeFocused() for iOS and make them log something
4. Use the custom control in a Content Page

### Expected Behavior
Overridden methods should be called every time the screen reader give the accessibility focus to the custom renderer, and so your message should be logged.

### Actual Behavior
Methods are never called, nothing is logged.

### Basic Information

- Version with issue: tried with 4.2.0.709249 and 4.4.0.0991537
- Last known good version:
- IDE: VS Mac 8.3.11 (build 1)
- Platform Target Frameworks: 
  - iOS:  
  - Android:  
- Android Support Library Version: 
- Nuget Packages: Xamarin.Forms, Xamarin.Essentials
- Affected Devices: Android and iOS

### Reproduction Link
[My project example](https://github.com/Poli97/CustomViewAccessibility) (in my example project called CustomViewAccessibility i try with ButtonRenderer)

**Comment by jgold6** [comment]
Issue here is that the are overriding methods that will be called on a native button, and a ButtonRenderer is not a subclass of the native button.

In a nutshell, you need to subclass UIKit.UIButton (iOS) and Android.Widget.Button (Android) and override those methods on the subclassed native button.
Then you need to override the CreateNativeControl method in the custom renderer and return an instance of your subclassed button. Then it works as expected.

Example custom renderers:

iOS:

```
public class IOSCustomView : ButtonRenderer
    {
        protected override void OnElementChanged(ElementChangedEventArgs<Button> e)
        {

            base.OnElementChanged(e);

            if (Control == null)
            {
                return;
            }

            Console.WriteLine("PIPPO created from IOS");
            Control.BackgroundColor = UIColor.Red;
			
        }

		protected override UIButton CreateNativeControl()
		{
			return new AccessibleButton();
		}


	}

	public class AccessibleButton : UIButton
	{
		
		public AccessibleButton() :base()
		{
			Console.WriteLine($"***** AccessibleButton created *****");
			
		}

		public override void AccessibilityElementDidBecomeFocused()
		{
			base.AccessibilityElementDidBecomeFocused();
			this.BackgroundColor = UIColor.Green;
			Console.WriteLine("PIPPO I am in focus");
		}

		public override void AccessibilityElementDidLoseFocus()
		{
			base.AccessibilityElementDidLoseFocus();
			this.BackgroundColor = UIColor.Red;
			Console.WriteLine("PIPPO I am NOT in focus");
		}
	}
```

Android: 

```
    public class AndroidCustomView : ButtonRenderer
    {
        Context context;

        public AndroidCustomView(Context context) : base(context)
        {
            this.context = context;
        }

        protected override void OnElementChanged(ElementChangedEventArgs<Xamarin.Forms.Button> e)
        {
            base.OnElementChanged(e);

            if (Control == null)
            {
                return;
            }

            Console.WriteLine("PIPPO created from Android");
            Control.SetBackgroundColor(Android.Graphics.Color.Red);
        }

		protected override Android.Widget.Button CreateNativeControl()
		{
			return new AccessibleButton(context);
		}

		
    }

	public class AccessibleButton : Android.Widget.Button
	{
		public AccessibleButton(Context context) : base(context)
		{
            Console.WriteLine($"***** AccessibleButton created *****");
        }

		public override void OnInitializeAccessibilityEvent(AccessibilityEvent e)
		{
			base.OnInitializeAccessibilityEvent(e);
            if (e.EventType == EventTypes.ViewAccessibilityFocused)
            {
                this.SetBackgroundColor(Android.Graphics.Color.Green);
                Console.WriteLine("PIPPO I am in focus");
            }
			else if (e.EventType == EventTypes.ViewAccessibilityFocusCleared)
			{
                this.SetBackgroundColor(Android.Graphics.Color.Red);
                Console.WriteLine("PIPPO I am in NOT in focus");
            }
        }
	}
```


If anything, this is a feature request. Xamarin.Forms would have to use UIButton and Android.Widget.Button subclasses instead of the default native buttons so that those accessibility methods can be overridden and raise events that the renderer can subscribe to (or some such).
